libs/ngx-sherfire/src/lib/firestore/derivable-firestore.service.ts

```diff
@@ -0,0 +1,136 @@
+import { Injectable, NgZone } from '@angular/core';
+import { Derivable, DerivableAtom, SettableDerivable, error, lens } from '@skunkteam/sherlock';
+import { DerivableCacheOptions, derivableCache, fromEventPattern } from '@skunkteam/sherlock-utils';
+import {
+    CollectionReference,
+    DocumentData,
+    DocumentReference,
+    DocumentSnapshot,
+    FirestoreError,
+    Query,
+    QueryDocumentSnapshot,
+    QuerySnapshot,
+    onSnapshot,
+    setDoc,
+} from 'firebase/firestore';
+import { NgxSherfireModule } from '../ngx-sherfire.module';
+
+@Injectable({ providedIn: NgxSherfireModule })
+export class DerivableFirestore {
+    constructor(private readonly zone: NgZone) {}
+
+    /**
+     * Build a derivable based on a Query or CollectionReference that returns a stream of updated QuerySnapshots that contain all
+     * QueryDocumentSnapshots matching the query.
```

Comment:
1. I believe that the word "_stream_" is too overloaded/technical to be meaningful here and also never used in relation to Sherlock's `Derivables`. Results are not streamed, but the resulting `Derivable` is "kept in sync with / updated from the server". Please change all uses of "_stream_" in this file.

2. You could also consider to combine the docs for all overloads into this single JSDoc section. Intellisense shows the first docs when invoking syntax completion in the IDE.

Reply:
I've rewritten the docs. Personally I prefer having separate docs for the Query and DocumentRef usecases. They work fine in VS Code code completion. Whenever you type `data$(` you get proper docs and can browse through the two overloads.
It keeps the docs simpler than having a single text that covers both use cases. But let me know if you still want me to rewrite to a single doc.